use public APIs through ember-modifier and upgrade dependencies

andreyfel opened the pull request:
Use public APIs: capabilities and setModifierManager.
Remove resolutions, as they are not needed.
Remove registering of the modifier in tests.
Update to latest.

jelhan commented on package.json:
```diff
@@ -21,44 +21,40 @@
+    "ember-cli-template-lint": "^1.0.0-beta.3",
+    "ember-cli-uglify": "^3.0.0",
     "ember-disable-prototype-extensions": "^1.1.3",
-    "ember-export-application-global": "^2.0.0",
```
Why did you removed `ember-export-application-global`. It's still part of default blueprint, isn't it?

andreyfel replied:
Oh, I thought that it is not used. If you think it is useful, I can bring it back.

jelhan replied:
To be honest, I don't know if it's needed but would feel more comfortable if we stick to default blueprints as much as possible to avoid unnecessary complications.

andreyfel replied:
This addon is supposed to be used for debugging. I think it is not needed in this addon, it is safe to delete it. But I can bring it back tomorrow if you still think it is needed.

andreyfel replied:
Brought it back

jelhan commented on addon/modifiers/style.js:
```diff
@@ -1,4 +1,4 @@
-import Ember from 'ember';
+import { setModifierManager, capabilities } from '@ember/modifier';
```
Is `ember-modifier-manager-polyfill` supporting that API? In it's documentation it's only using `Ember._setModifierManager()`. If not I would be also fine with dropping support for Ember < 3.13.

andreyfel replied:
It is supposed to work with older Ember as `ember-modifier-manager-polyfill` polyfills those APIs. I think the documentation is out of date. I took as a reference https://github.com/emberjs/ember-render-modifiers which is also using the polyfill.

jelhan replied:
In that case we might want to add Ember 2.18 back to the supported versions? Only if it does not cause any additional work. This one would be a breaking change anyways cause it drops support for node 6.

andreyfel replied:
Returned 2.18 to the ember-try and travis configs

jelhan commented on addon/modifiers/style.js:
```diff
@@ -59,8 +59,10 @@ function setStyles(element, newStyles, oldStyles) {
-export default Ember._setModifierManager(
+export default setModifierManager(
   () => ({
+    capabilities: capabilities('3.13'),
```
How does this work together with the polyfill? If the `ember-modifier-manager-polyfill` supports that API at all. As already that above, dropping support for Ember < 3.13 would be also fine for me.

andreyfel replied:
We don't need to drop Ember < 3.13 as `ember-modifier-manager-polyfill` polyfills capabilities API as well.

jelhan commented on package.json:
```diff
@@ -21,44 +21,40 @@
     "configPath": "tests/dummy/config"
-  },
-  "resolutions": {
-    "qunit": "^2.9.1"
```
Resolutions were required cause `ember-qunit` defaults to an old version of `qunit`, which was violating a strict content security policy (CSP). But `ember-qunit` has finally updated it's `qunit` version. So not required anymore. :+1:

andreyfel commented:
Tests were failing because useYarn got lost from the ember-try config during ember-cli-update. I brought it back and now tests pass for everything but Ember 3.4. That is weird and they are supposed to work, trying to figure out what is missing there.

andreyfel commented:
I don't know why it can't resolve "style" as a modifier for 3.4...
@rwjblue do you have any idea?
It says "Error: Compile Error style is not a modifier: Helpers may not be used in the element form."
It passes for all other testable Ember releases.

kategengler commented on .travis.yml:
```diff
@@ -49,19 +49,12 @@ jobs:
     - env: EMBER_TRY_SCENARIO=ember-beta
     - env: EMBER_TRY_SCENARIO=ember-canary
     - env: EMBER_TRY_SCENARIO=ember-default-with-jquery
 
```
Lost the yarn parts of .travis.yml here

andreyfel replied:
@kategengler do you think it is needed? I see that it works even without it.

kategengler replied:
Yes. The before_install is probably unnecessary, it may only make sure you get the latest yarn version. The install step here is important because it adds `--no-lockfile` option that ensures dependencies float, without it your 'Floating Dependencies' stage is the same as part of the 'Tests' stage (see https://travis-ci.org/jelhan/ember-style-modifier/jobs/620199516#L480).

andreyfel replied:
Good to know! Thanks! Returned that bit to the travis config.

andreyfel commented:
I 've checked versions between 3.4 and 3.8. In 3.6 it started working. I realized it is something related to the changes made in 3.6. The most significant one was "Native Class Constructor Update". I've tried to add `ember-native-class-polyfill` and it helped!

andreyfel commented:
I've also changed the addon to be based on https://github.com/ember-modifier/ember-modifier (which will likely become a standard in modifiers world) instead of implementing our own modifier manager.
If you don't like this bit, I can drop this commit.

andreyfel commented:
@kategengler helped to solve the mystery! `ember-cli-update` changed the dummy app declaration to be a native class. It caused the app to miss the Resolver because in 3.4 the order of assignment is inverted without `ember-native-class-polyfill`. So, it is ok to have the polyfill in dev deps as it affects only the tests.

jelhan commented:
> I've also changed the addon to be based on https://github.com/ember-modifier/ember-modifier (which will likely become a standard in modifiers world) instead of implementing our own modifier manager.
> If you don't like this bit, I can drop this commit.

I like it. Makes the implementation much easier to understand.

> @kategengler helped to solve the mystery! ember-cli-update changed the dummy app declaration to be a native class. It caused the app to miss the Resolver because in 3.4 the order of assignment is inverted without ember-native-class-polyfill. So, it is ok to have the polyfill in dev deps as it affects only the tests.

I'm not quite sure but isn't this creating the risk of tests passing only because the `ember-native-class-polyfill` is available in testing environment? The modifier is also using a native class. The tests aren't telling us currently if it would still work in Ember 3.4 without the native class polyfill being available, do they?

andreyfel commented:
The  `ember-native-class-polyfill` is needed if you use class fields with default values. The modifier itself doesn't use them. I removed the polyfill and reverted transition to native class syntax for Route and Application. All the tests passed even for Ember 2.18.

jelhan commented:
@andreyfel Thanks a lot for working on this one!

andreyfel commented:
@jelhan great! Thanks! Will you publish new release?

jelhan commented:
> @jelhan great! Thanks! Will you publish new release?

Yeah. Will try to do it today but not sure if I can manage to do so. Like to add `release-it` and `lerna-changelog` before to automate the process. Being in vacation starting from tomorrow. If I can't do it today, I will do it at the beginning of next week for sure.

jelhan commented:
@andreyfel Just released `v0.5.0`.

andreyfel commented:
Awesome! Thank you so much!
